src/roadmap/v1/lifecycle/app_streams.py

```diff
@@ -240,18 +168,47 @@ async def get_app_stream_names(
     }
 
 
-## Relevant ##
-relevant = APIRouter(
-    prefix="/relevant/lifecycle/app-streams",
-    tags=["Relevant", "App Streams"],
-)
+class AppStreamKey:
+    """Wraps AppStreamEntity objects in"""
 
+    def __init__(self, app_stream_entity: AppStreamEntity, name: str):
+        self.app_stream_entity = app_stream_entity
+        self.name = name
 
-@relevant.get("", response_model=RelevantAppStreamsResponse)
-async def get_relevant_app_streams(
+    def __hash__(self):
+        return (
+            self.name,
+            self.app_stream_entity.display_name,
+            self.app_stream_entity.application_stream_name,
+            self.app_stream_entity.os_major,
+            self.app_stream_entity.os_minor,
+            self.app_stream_entity.start_date,
+            self.app_stream_entity.end_date,
+            str(self.app_stream_entity.impl),
+            self.app_stream_entity.rolling,
+        ).__hash__()
+
+    def __eq__(self, other):
+        return isinstance(other, AppStreamKey) and self.__hash__() == other.__hash__()
+
+
+def related_app_streams(app_streams: list[AppStreamKey]) -> list[AppStreamKey]:
+    """Return unique list of related apps that do not appear in app_streams."""
+    relateds = set()
+    for app_stream_key in app_streams:
+        for app in APP_STREAM_MODULES_PACKAGES:
+            if app.match_name == app_stream_key.app_stream_entity.match_name:
+                if streams_lt(app_stream_key.app_stream_entity.stream, app.stream):
+                    if app.end_date is None or app.end_date > date.today():
+                        relateds.add(AppStreamKey(app, app_stream_key.name))
+    return relateds - set(app_streams)
```

Comment:
```suggestion
    return relateds.difference(app_streams)
```
Saves on converting the `app_streams` list into a set. (In general, I like to avoid using operator syntax for sets too because they aren't self-documenting.)

Reply:
fixed. (thanks, didn't know this one!)